src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,72 @@
+import java.util.Arrays;
+
 public class StringManipulation implements StringManipulationInterface {
 
+    private String myString;
     @Override
     public String getString() {
-        return null;
+        return myString;
     }
 
     @Override
     public void setString(String string) {
+        myString = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        int countWords = myString.split("\\w+").length;
+        if (myString.charAt(0) == ' ') countWords --;
+        return countWords;
     }
 
     @Override
     public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+        if (n > myString.length()) {
+            throw new IndexOutOfBoundsException ("n is greater than the string length");
+        }
+        if (n <= 0) {
+            throw  new IllegalArgumentException("n can not be less than or equal to zero");
+        }
+        String str = "";
+        for (int i = 0; i < myString.length(); i++) {
+            if (i%n != n-1) {
+                str += myString.charAt(i);
+            } else {
+                if (maintainSpacing) str += " ";
+            }
+        }
+        return str;
     }
 
     @Override
-    public String[] getSubStrings(int startWord, int endWord) {
-        return null;
+    public String[] getSubStrings(int startWord, int endWord){
+        if (startWord <= 0 || endWord <= 0 || startWord > endWord) throw new IllegalArgumentException("invalid input for index");
```

Comment:
I also did write the same things as you before I get a reviewer to say that it would be better for user if you can make the case startWord > endWord with a different if case so that you can throw a more specific error for that case.